homeassistant/components/bmw_connected_drive/binary_sensor.py

```diff
@@ -142,13 +126,18 @@ def update(self):
                 result["check_control_messages"] = cbs_list
             else:
                 result["check_control_messages"] = "OK"
-        elif self._attribute == "charging_status":
+        # device class power: On means power detected, Off means no power
+        if self._attribute == "charging_status":
+            self._attr_is_on = vehicle_state.charging_status in [ChargingState.CHARGING]
             result["charging_status"] = vehicle_state.charging_status.value
             result["last_charging_end_result"] = vehicle_state.last_charging_end_result
-        elif self._attribute == "connection_status":
+        # device class plug: On means device is plugged in,
+        #                    Off means device is unplugged
+        if self._attribute == "connection_status":
```

Comment:
All if's but the first should be elif's `self._attribute` can only match 1 of them